src/AzureAuth/CommandMain.cs

```diff
@@ -444,6 +494,21 @@ private int GetToken()
             return 0;
         }
 
+        private void SendAuthFlowTelemetryEvents(AuthFlowResult[] results)
+        {
+            if (results != null)
+            {
+                Parallel.ForEach(results, result =>

```

Comment:
Also, the results gathered are not that huge in number, so there might not be a lot of benefit to do parallel foreach here.  Sometimes creating overhead of creating multiple threads for a small collection can be larger than any benefit achieved by processing it in parallel. The processing of each result is not very expensive since you just create an event data object from a result object.

Reply:
I think it's a good to have even though it doesn't improve the performance much here.